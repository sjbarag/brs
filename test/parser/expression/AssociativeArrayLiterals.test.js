const BrsError = require("../../../lib/Error");
const { Lexeme, BrsTypes, Parser } = require("brs");
const { BrsString, Int32 } = BrsTypes;

const { EOF } = require("../ParserTests");

describe("parser associative array literals", () => {
    afterEach(() => BrsError.reset());

    describe("empty associative arrays", () => {
        test("on one line", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftBrace, text: "{", line: 1 },
                { kind: Lexeme.RightBrace, text: "}", line: 1 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        test("on multiple lines", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftBrace, text: "{", line: 1 },
                { kind: Lexeme.Newline, text: "\n", line: 1 },
                { kind: Lexeme.Newline, text: "\n", line: 2 },
                { kind: Lexeme.Newline, text: "\n", line: 3 },
                { kind: Lexeme.Newline, text: "\n", line: 4 },
                { kind: Lexeme.Newline, text: "\n", line: 5 },
                { kind: Lexeme.Newline, text: "\n", line: 6 },
                { kind: Lexeme.RightBrace, text: "}", line: 7 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });

    describe("filled arrays", () => {
        test("on one line", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftBrace, text: "{", line: 1 },
                { kind: Lexeme.Identifier, text: "foo", line: 1 },
                { kind: Lexeme.Colon, text: ":", line: 1 },
                { kind: Lexeme.Integer, text: "1", line: 1, literal: new Int32(1) },
                { kind: Lexeme.Comma, text: ",", line: 1 },
                { kind: Lexeme.Identifier, text: "bar", line: 1 },
                { kind: Lexeme.Colon, text: ":", line: 1 },
                { kind: Lexeme.Integer, text: "2", line: 1, literal: new Int32(2) },
                { kind: Lexeme.Comma, text: ",", line: 1 },
                { kind: Lexeme.Identifier, text: "baz", line: 1 },
                { kind: Lexeme.Colon, text: ":", line: 1 },
                { kind: Lexeme.Integer, text: "3", line: 1, literal: new Int32(3) },
                { kind: Lexeme.RightBrace, text: "}", line: 1 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        test("on multiple lines with commas", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftBrace, text: "{", line: 1 },
                { kind: Lexeme.Newline, text: "\n", line: 1 },
                { kind: Lexeme.Identifier, text: "foo", line: 2 },
                { kind: Lexeme.Colon, text: ":", line: 2 },
                { kind: Lexeme.Integer, text: "1", line: 2, literal: new Int32(1) },
                { kind: Lexeme.Comma, text: ",", line: 2 },
                { kind: Lexeme.Newline, text: "\n", line: 2 },
                { kind: Lexeme.Identifier, text: "bar", line: 3 },
                { kind: Lexeme.Colon, text: ":", line: 3 },
                { kind: Lexeme.Integer, text: "2", line: 3, literal: new Int32(2) },
                { kind: Lexeme.Comma, text: ",", line: 3 },
                { kind: Lexeme.Newline, text: "\n", line: 3 },
                { kind: Lexeme.Identifier, text: "baz", line: 4 },
                { kind: Lexeme.Colon, text: ":", line: 4 },
                { kind: Lexeme.Integer, text: "3", line: 4, literal: new Int32(3) },
                { kind: Lexeme.Newline, text: "\n", line: 5 },
                { kind: Lexeme.RightBrace, text: "}", line: 6 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        test("on multiple lines without commas", () => {
            let parsed = Parser.parse([
                { kind: Lexeme.Identifier, text: "_", line: 1 },
                { kind: Lexeme.Equal, text: "=", line: 1 },
                { kind: Lexeme.LeftBrace, text: "{", line: 1 },
                { kind: Lexeme.Newline, text: "\n", line: 1 },
                { kind: Lexeme.Identifier, text: "foo", line: 2 },
                { kind: Lexeme.Colon, text: ":", line: 2 },
                { kind: Lexeme.Integer, text: "1", line: 2, literal: new Int32(1) },
                { kind: Lexeme.Newline, text: "\n", line: 2 },
                { kind: Lexeme.Identifier, text: "bar", line: 3 },
                { kind: Lexeme.Colon, text: ":", line: 3 },
                { kind: Lexeme.Integer, text: "2", line: 3, literal: new Int32(2) },
                { kind: Lexeme.Newline, text: "\n", line: 3 },
                { kind: Lexeme.Identifier, text: "baz", line: 4 },
                { kind: Lexeme.Colon, text: ":", line: 4 },
                { kind: Lexeme.Integer, text: "3", line: 4, literal: new Int32(3) },
                { kind: Lexeme.Newline, text: "\n", line: 5 },
                { kind: Lexeme.RightBrace, text: "}", line: 6 },
                EOF
            ]);

            expect(BrsError.found()).toBeFalsy();
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });

    it("allows a mix of quoted and unquoted keys", () => {
        let parsed = Parser.parse([
            { kind: Lexeme.Identifier, text: "_", line: 1 },
            { kind: Lexeme.Equal, text: "=", line: 1 },
            { kind: Lexeme.LeftBrace, text: "{", line: 1 },
            { kind: Lexeme.Newline, text: "\n", line: 1 },
            { kind: Lexeme.String, text: "foo", line: 2, literal: new BrsString("foo") },
            { kind: Lexeme.Colon, text: ":", line: 2 },
            { kind: Lexeme.Integer, text: "1", line: 2, literal: new Int32(1) },
            { kind: Lexeme.Comma, text: ",", line: 2 },
            { kind: Lexeme.Newline, text: "\n", line: 2 },
            { kind: Lexeme.Identifier, text: "bar", line: 3 },
            { kind: Lexeme.Colon, text: ":", line: 3 },
            { kind: Lexeme.Integer, text: "2", line: 3, literal: new Int32(2) },
            { kind: Lexeme.Comma, text: ",", line: 3 },
            { kind: Lexeme.Newline, text: "\n", line: 3 },
            { kind: Lexeme.String, text: "requires-hyphens", line: 4, literal: new BrsString("requires-hypens") },
            { kind: Lexeme.Colon, text: ":", line: 4 },
            { kind: Lexeme.Integer, text: "3", line: 4, literal: new Int32(3) },
            { kind: Lexeme.Newline, text: "\n", line: 5 },
            { kind: Lexeme.RightBrace, text: "}", line: 6 },
            EOF
        ]);

        expect(BrsError.found()).toBeFalsy();
        expect(parsed).toBeDefined();
        expect(parsed).not.toBeNull();
        expect(parsed).toMatchSnapshot();
    });
});
