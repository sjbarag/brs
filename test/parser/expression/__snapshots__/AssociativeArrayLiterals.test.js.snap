// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser associative array literals allows a mix of quoted and unquoted keys 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": AALiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "}",
      },
      "elements": Array [
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "foo",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 1,
            },
          },
        },
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "bar",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 2,
            },
          },
        },
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "requires-hyphens",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 3,
            },
          },
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "{",
      },
      "type": "AALiteral",
    },
  },
]
`;

exports[`parser associative array literals allows separating properties with colons 1`] = `
Array [
  Function {
    "func": Function {
      "body": Block {
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "file": undefined,
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "statements": Array [
          Assignment {
            "name": Object {
              "isReserved": false,
              "kind": "Identifier",
              "literal": undefined,
              "location": Object {
                "end": Object {
                  "column": -9,
                  "line": -9,
                },
                "start": Object {
                  "column": -9,
                  "line": -9,
                },
              },
              "text": "person",
            },
            "tokens": Object {
              "equals": Object {
                "isReserved": false,
                "kind": "Equal",
                "literal": undefined,
                "location": Object {
                  "end": Object {
                    "column": -9,
                    "line": -9,
                  },
                  "start": Object {
                    "column": -9,
                    "line": -9,
                  },
                },
                "text": "=",
              },
            },
            "type": "Assignment",
            "value": AALiteral {
              "close": Object {
                "isReserved": false,
                "kind": "RightBrace",
                "literal": undefined,
                "location": Object {
                  "end": Object {
                    "column": -9,
                    "line": -9,
                  },
                  "start": Object {
                    "column": -9,
                    "line": -9,
                  },
                },
                "text": "}",
              },
              "elements": Array [
                Object {
                  "name": BrsString {
                    "kind": 3,
                    "value": "name",
                  },
                  "value": Literal {
                    "_location": Object {
                      "end": Object {
                        "column": -9,
                        "line": -9,
                      },
                      "start": Object {
                        "column": -9,
                        "line": -9,
                      },
                    },
                    "type": "Literal",
                    "value": BrsString {
                      "kind": 3,
                      "value": "Bob",
                    },
                  },
                },
                Object {
                  "name": BrsString {
                    "kind": 3,
                    "value": "age",
                  },
                  "value": Literal {
                    "_location": Object {
                      "end": Object {
                        "column": -9,
                        "line": -9,
                      },
                      "start": Object {
                        "column": -9,
                        "line": -9,
                      },
                    },
                    "type": "Literal",
                    "value": Int32 {
                      "kind": 4,
                      "value": 3,
                    },
                  },
                },
              ],
              "open": Object {
                "isReserved": false,
                "kind": "LeftBrace",
                "literal": undefined,
                "location": Object {
                  "end": Object {
                    "column": -9,
                    "line": -9,
                  },
                  "start": Object {
                    "column": -9,
                    "line": -9,
                  },
                },
                "text": "{",
              },
              "type": "AALiteral",
            },
          },
        ],
        "type": "Block",
      },
      "endKeyword": Object {
        "isReserved": false,
        "kind": "EndSub",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "end sub",
      },
      "keyword": Object {
        "isReserved": true,
        "kind": "Sub",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "sub",
      },
      "parameters": Array [],
      "returns": 11,
      "type": "Expr_Function",
    },
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "main",
    },
    "type": "Stmt_Function",
  },
]
`;

exports[`parser associative array literals empty associative arrays on multiple lines 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": AALiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "}",
      },
      "elements": Array [],
      "open": Object {
        "isReserved": false,
        "kind": "LeftBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "{",
      },
      "type": "AALiteral",
    },
  },
]
`;

exports[`parser associative array literals empty associative arrays on one line 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": AALiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "}",
      },
      "elements": Array [],
      "open": Object {
        "isReserved": false,
        "kind": "LeftBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "{",
      },
      "type": "AALiteral",
    },
  },
]
`;

exports[`parser associative array literals filled arrays on multiple lines with commas 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": AALiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "}",
      },
      "elements": Array [
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "foo",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 1,
            },
          },
        },
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "bar",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 2,
            },
          },
        },
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "baz",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 3,
            },
          },
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "{",
      },
      "type": "AALiteral",
    },
  },
]
`;

exports[`parser associative array literals filled arrays on multiple lines without commas 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": AALiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "}",
      },
      "elements": Array [
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "foo",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 1,
            },
          },
        },
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "bar",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 2,
            },
          },
        },
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "baz",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 3,
            },
          },
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "{",
      },
      "type": "AALiteral",
    },
  },
]
`;

exports[`parser associative array literals filled arrays on one line 1`] = `
Array [
  Assignment {
    "name": Object {
      "isReserved": false,
      "kind": "Identifier",
      "literal": undefined,
      "location": Object {
        "end": Object {
          "column": -9,
          "line": -9,
        },
        "start": Object {
          "column": -9,
          "line": -9,
        },
      },
      "text": "_",
    },
    "tokens": Object {
      "equals": Object {
        "isReserved": false,
        "kind": "Equal",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "=",
      },
    },
    "type": "Assignment",
    "value": AALiteral {
      "close": Object {
        "isReserved": false,
        "kind": "RightBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "}",
      },
      "elements": Array [
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "foo",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 1,
            },
          },
        },
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "bar",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 2,
            },
          },
        },
        Object {
          "name": BrsString {
            "kind": 3,
            "value": "baz",
          },
          "value": Literal {
            "_location": Object {
              "end": Object {
                "column": -9,
                "line": -9,
              },
              "start": Object {
                "column": -9,
                "line": -9,
              },
            },
            "type": "Literal",
            "value": Int32 {
              "kind": 4,
              "value": 3,
            },
          },
        },
      ],
      "open": Object {
        "isReserved": false,
        "kind": "LeftBrace",
        "literal": undefined,
        "location": Object {
          "end": Object {
            "column": -9,
            "line": -9,
          },
          "start": Object {
            "column": -9,
            "line": -9,
          },
        },
        "text": "{",
      },
      "type": "AALiteral",
    },
  },
]
`;
